dev/com.ibm.ws.microprofile.metrics.cdi.3.0/src/io/astefanutti/metrics/cdi30/MetricResolver.java

```diff
@@ -0,0 +1,481 @@
+            return ((Metered) annotation).description();
+        else if (Timed.class.isInstance(annotation))
+            return ((Timed) annotation).description();
+        else if (SimplyTimed.class.isInstance(annotation))
+            return ((SimplyTimed) annotation).description();
+        else
+            throw new IllegalArgumentException("Unsupported Metrics forMethod [" + annotation.getClass().getName() + "]");
+    }
+
+    private MetricType getType(Annotation annotation) {
+        if (Counted.class.isInstance(annotation))
+            return MetricType.COUNTER;
+        else if (ConcurrentGauge.class.isInstance(annotation))
+            return MetricType.CONCURRENT_GAUGE;
+        else if (Gauge.class.isInstance(annotation))
+            return MetricType.GAUGE;
+        else if (Metered.class.isInstance(annotation))
+            return MetricType.METERED;
+        else if (Timed.class.isInstance(annotation))
+            return MetricType.TIMER;
+        else if (SimplyTimed.class.isInstance(annotation))
+            return MetricType.SIMPLE_TIMER;
+        else
+            throw new IllegalArgumentException("Unsupported Metrics forMethod [" + annotation.getClass().getName() + "]");
+    }
+
+    private String getUnit(Annotation annotation) {
+        if (Counted.class.isInstance(annotation))
+            return ((Counted) annotation).unit();
+        else if (ConcurrentGauge.class.isInstance(annotation))
+            return MetricUnits.NONE;
+        else if (Gauge.class.isInstance(annotation))
+            return ((Gauge) annotation).unit();
+        else if (Metered.class.isInstance(annotation))
+            return ((Metered) annotation).unit();
+        else if (Timed.class.isInstance(annotation))
+            return ((Timed) annotation).unit();
+        else if (SimplyTimed.class.isInstance(annotation))
+            return ((SimplyTimed) annotation).unit();
+        else
+            throw new IllegalArgumentException("Unsupported Metrics forMethod [" + annotation.getClass().getName() + "]");
+    }
+
+    /**
+     * Checks whether the metric should be re-usable
+     */
+    private <T extends Annotation> boolean checkReusable(MetricResolver.Of<T> of) {
+
+        // If the metric has been registered before (eg. metrics found in RequestScoped beans),
+        // we don't need to worry about re-usable
+        if (!of.isInitialDiscovery()) {
+            return true;
+        }
+
+        String name = of.metadata().getName();
+        String[] tags = of.tags();
+        MetricID metricID = new MetricID(name, Utils.tagsToTags(tags));
+
+        Metadata existingMetadata = registry.getMetadata(name);
+
+        /*
+         * If the current metadata does not equal an existing metadata (of matching name).. - Then we can NOT re-use.
+         * Type check is conducted when registering/retrieving metric.
+         */
+        if ((existingMetadata != null && !existingMetadata.equals(of.metadata()))) {
```

Comment:
Removal of reusability only checks if metadata is different.
There also use to exist code that retrieved the `reusable` value from the annotation. But that has been removed as is not apparant in this diff.